satpy/readers/modis_l1b.py

```diff
@@ -62,130 +90,164 @@ class HDFEOSBandReader(HDFEOSBaseFileReader):
            "Q": 250,
            "H": 500}
 
-    def __init__(self, filename, filename_info, filetype_info):
+    res_to_possible_variable_names = {
+        1000: ['EV_250_Aggr1km_RefSB',
+               'EV_500_Aggr1km_RefSB',
+               'EV_1KM_RefSB',
+               'EV_1KM_Emissive'],
+        500: ['EV_250_Aggr500_RefSB',
+              'EV_500_RefSB'],
+        250: ['EV_250_RefSB'],
+    }
+
+    def __init__(self, filename, filename_info, filetype_info, mask_saturated=True, **kwargs):
```

Comment:
What are the kwargs for?

Reply:
The kwargs are actually needed if this reader (file handler) is loaded at the same time as other readers. This is an unfortunate side effect of how the Scene handles reader keyword arguments and we've done it in other readers. The Scene, when given a dictionary of keyword arguments, will pass them to all readers being loaded. If a file handler/reader doesn't have `**kwargs` then any unrecognized keyword arguments will raise an exception.

Recently @gerritholl added the ability to specify the exact reader you want to pass keyword arguments to from the Scene `__init__` so maybe this isn't required anymore, but I'd be nervous about deprecating it too fast. I could remove it for this reader with the understanding that I only need to handle `mask_saturated` and the assumption that the base HDFEOS file handler has no other keyword arguments. I could then remove the `**kwargs` that I added to the base HDFEOS file handler above.

Reply:
So I looked at this more and I think we need this in the other file handlers. The base and geo file handlers in `hdfeos.py` will also both receive the `mask_saturated=True` keyword argument and won't know what to do with it. The easiest way to capture that and ignore it is to use `**kwargs`.